### PR TITLE
vttablet: Cleanup unused db version string

### PR DESCRIPTION
**Pull request by dbussink**
We're reading the database version string from the `mysqld` binary, but we subsequently never use it anywhere.

In case Vitess is used with containers but with a separate MySQL container, this check is also not correct as it would report whatever is bundled with Vitess, not the MySQL version that actually runs.

Since we don't need to have it and because of the above reason it's better that we remove it here.

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on the CI
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go/json2/unmarshal.go [hunk 1]
```diff
@@ -33,7 +33,8 @@ var carriageReturn = []byte("\n")
 // efficient and should not be used for high QPS operations.
 func Unmarshal(data []byte, v any) error {
 	if pb, ok := v.(proto.Message); ok {
-		return annotate(data, protojson.Unmarshal(data, pb))
+		opts := protojson.UnmarshalOptions{DiscardUnknown: true}
+		return annotate(data, opts.Unmarshal(data, pb))
```
Comment (dbussink) [inline go/json2/unmarshal.go:37]: This behavior more follows how protobuf itself works, which allows for fields to be removed and it ignores it in those cases.

File: proto/topodata.proto [hunk 2]
```diff
@@ -157,16 +157,14 @@ message Tablet {
   //
   vttime.Time primary_term_start_time = 14;
 
-  // db_server_version represents the database version used by the tablet.
-  string db_server_version = 15;
-
```
Comment (frouioui) [inline proto/topodata.proto:162]: Is this removal worth a note in the release notes?

**Review by frouioui — APPROVED** [review]

File: proto/topodata.proto (hunk 2 above)
Comment (dbussink) [reply proto/topodata.proto:162]: I don't know? Do we generally add release notes for what are (afaik?) internal fields for Vitess?

Comment (deepthi) [reply proto/topodata.proto:162]: We don't need this in the release notes.

**Review by deepthi — APPROVED** [review]

**Pull request merged**